src/screens/Users/AddOrEditUser.js

```diff
@@ -144,48 +472,102 @@ class AddOrEditUser extends React.PureComponent<Props, State> {
     this.setState({ visibleModal: '' });
   };
 
-  verifyEmail = () => {
-    this.setState({ verifyingField: 'email' });
-  };
-
-  verifyPhone = () => {
-    this.setState({ verifyingField: 'phone' });
+  verifyField = (verifyingField) => {
+    this.setState({ verifyingField });
   };
 
   onCloseVerification = () => {
-    this.setState({ verifyingField: null });
+    const { verifyingField } = this.state;
+    this.setState({ verifyingField: null, verifiedModalField: verifyingField });
   };
 
-  getAlertData = (dataType: string) => {
-    return ({
-      title: `Change ${dataType}`,
-      message: `You will have to re-verify your ${dataType} after changing it`,
+  changeField = () => {
+    const { updateUser, user } = this.props;
+    const { cautionModalField, value } = this.state;
+    if (cautionModalField === 'phone') {
+      const { phone: { input, selector } } = value;
+      const formattedPhone = input ? `+${selector.callingCode}${input}` : null;
+      updateUser(user.walletId, { phone: formattedPhone });
+    } else if (cautionModalField === 'email') {
+      updateUser(user.walletId, { email: value.email });
+    }
+    this.setState({ cautionModalField: null });
+  }
+
+  onDismissCautionModal = () => {
+    this.setState({
+      cautionModalField: null,
+      value: getInitialValue(this.props.user),
     });
-  };
+  }
+
+  renderInsight = () => {
+    const {
+      verificationNoteDismissed,
+      privacyInsightDismissed,
+      dismissPrivacyInsight,
+      dismissVerificationNote,
+      user,
+      goToInvitationFlow,
+      isPillarRewardCampaignActive,
+    } = this.props;
+    const {
+      isEmailVerified, isPhoneVerified, email, phone,
+    } = user;
+    if (isPhoneVerified || isEmailVerified) {
+      return (
+        <InviteBanner
+          title="Pillar is social"
+          onInvitePress={goToInvitationFlow}
+          isReferralActive={isPillarRewardCampaignActive}
+        />
+      );
+    }
+    if (email || phone) {
+      if (verificationNoteDismissed) return null;
+      return (
+        <Note
+          note="Verification is required to prevent scammers activity. We care for our users privacy."
+          containerStyle={{ marginHorizontal: spacing.layoutSides }}
+          onClose={dismissVerificationNote}
+        />
+      );
+    }
+    if (privacyInsightDismissed) return null;
+    return (
+      <InsightWithButton
+        title="All your data is private"
+        description="We care for our users' privacy. We don't support scammers.
+                     We need to make sure you're a genuine user in order to get
+                     rewarded with PLR tokens for inviting friends."
```

Comment:
This text should be based on `isPillarRewardCampaignActive`.
We should not mention reward in PLR if `isPillarRewardCampaignActive` is `false`.